### PR TITLE
OJ-36861-setup-jf-ingest-for-github

### PR DESCRIPTION
**Pull request by gavinpitt-jf**
### Description

Set up JF Ingest Github GQL for all companies that are already running the original Agent version of the GQL ingest. I tested this with orthogonal-networks in an "end to end" test; I ran an Agent run for our git instance, imported it to `stg-looking-glass` via `git_import`, and then ran the Agent again. I verified that the first Agent run and submission created a few PRs and commits and branches, and then verified that the second run didn't create anything new. 

This PR contains initial groundwork that will be helpful as we roll out ADO in the Agent (via `jf_ingest`)

### CODE REVIEW

File: jf_agent/config_file_reader.py [hunk 1]
```diff
@@ -308,12 +315,50 @@ def _get_git_config_from_yaml(yaml_config) -> List[GitConfig]:
...
+                token=git_creds['github_token'],
+                base_url=config.git_url,
+                verify=not skip_ssl_verification,
+            )
+        if config.git_provider == GL_PROVIDER:
+            return None
+
+    except Exception as e:
+        logging_helper.log_standard_error(
+            logging.ERROR, msg_args=[config.git_provider, e], error_code=2101, exc_info=True,
```
Comment (andra-jellyfish) [inline jf_agent/config_file_reader.py:342]: What does this error code map to?

File: jf_agent/config_file_reader.py [hunk 2]
```diff
@@ -380,6 +425,79 @@ def get_ingest_config(config: ValidatedConfig, creds, endpoint_jira_info: dict)
...
+        pull_from = _make_datetimes_timezone_aware(endpoint_git_instance_info['pull_from'])
+        git_configs.append(
+            JFIngestGitConfig(
+                company_slug=company_slug,
+                instance_slug=instance_slug,
+                instance_file_key=endpoint_git_instance_info['key'],
+                git_provider=agent_git_config.git_provider,
+                git_auth_config=jf_ingest_git_auth_config,
+                url=agent_git_config.git_url,
+                jf_options=jf_options,
```
Comment (andra-jellyfish) [inline jf_agent/config_file_reader.py:486]: is this a typo?

File: jf_agent/config_file_reader.py (hunk 1 above)
Comment (gavinpitt-jf) [reply jf_agent/config_file_reader.py:342]: This maps to "Failed to Connect", the lookup table is in the `logging_helper.py` [file in JF Ingest](https://github.com/Jellyfish-AI/jf_ingest/blob/491889ffeddc1b01aa95930bd6863e9951af64ae/jf_ingest/logging_helper.py#L101).

These Error codes were a part of JF Agent a LONG time ago, and as far as I know they aren't used meaningfully anywhere. Infact I think they do more harm than good by obscuring the code for no benefit. They're around as a legacy thing and should probably get revamped or removed

File: jf_agent/config_file_reader.py [hunk 3]
```diff
@@ -380,6 +425,79 @@ def get_ingest_config(config: ValidatedConfig, creds, endpoint_jira_info: dict)
...
+        agent_git_config: GitConfig = agent_git_config
+
+        if is_mult_git_mode:
+            instance_slug = agent_git_config.git_instance_slug
+            endpoint_git_instance_info = endpoint_git_instances_info.get(instance_slug)
+            instance_creds = creds.git_instance_to_creds.get(instance_slug)
+        else:
+            # If there's only one git config set, then this is "single git" mode.
+            # The instance slug is likely not to be provided
+            endpoint_git_instance_info = list(endpoint_git_instances_info.values())[0]
```
Comment (jruel4) [inline jf_agent/config_file_reader.py:440]: Q: Is this a result of having a "legacy" mode of sending things (non-GQL/ADO), as well as the new code? Are there any failure modes we're concerned about here?

File: jf_agent/git/utils.py [hunk 4]
```diff
@@ -4,6 +4,17 @@
...
+'''
+
+    Constants
+
+'''
+BBS_PROVIDER = 'bitbucket_server'
+BBC_PROVIDER = 'bitbucket_cloud'
+GH_PROVIDER = 'github'
+GL_PROVIDER = 'gitlab'
+PROVIDERS = [GL_PROVIDER, GH_PROVIDER, BBS_PROVIDER, BBC_PROVIDER]
```
Comment (jruel4) [inline jf_agent/git/utils.py:16]: Nit: Probably better do a tuple here to keep immutable unless we change this.

File: jf_agent/config_file_reader.py (hunk 3 above)
Comment (jruel4) [reply jf_agent/config_file_reader.py:440]: Or, now re-reading, is this not a requirement if the custom only provides a single git instance in their `config.yaml`?

**Review by jruel4 — APPROVED** [review]
LGTM pending a couple of questions

File: jf_agent/config_file_reader.py (hunk 3 above)
Comment (gavinpitt-jf) [reply jf_agent/config_file_reader.py:440]: This logic is used in a few places around the Agent: https://github.com/Jellyfish-AI/jf_agent/blob/08cfab5d49cd7f69d743519184aa9c31a8553b8f/jf_agent/main.py#L723

**Pull request merged**